apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/ContractServiceDefault.java

```diff
@@ -244,4 +248,66 @@ public Optional<File> getLogoFile() {
 
         return Optional.empty();
     }
+
+
+    @Override
+    public void uploadAggregatesCsv(OnboardingWorkflow onboardingWorkflow) {
+
+        try {
+            Onboarding onboarding = onboardingWorkflow.getOnboarding();
+            Path filePath = Files.createTempFile("tempfile", ".csv");
+            File csv = generateCsv(onboarding.getAggregates(), filePath);
+            final String path = String.format("%s%s/%s", azureStorageConfig.aggregatesPath(), onboarding.getId(),
+                    onboarding.getProductId());
+            final String filename = "aggregates.csv";
+            azureBlobClient.uploadFile(path, filename, Files.readAllBytes(csv.toPath()));
+        } catch (IOException e) {
+            throw new GenericOnboardingException(String.format("Can not load aggregates CSV, message: %s", e.getMessage()));
+        }
+    }
+
+    private File generateCsv(List<AggregateInstitution> institutions, Path filePath) {
+        String[] headers = {
+                "Ragione Sociale", "PEC", "Codice Fiscale", "P.IVA",
+                "Sede legale - Indirizzo", "Sede legale - Citta'", "Sede legale - Provincia (Sigla)",
+                "Codice IPA", "AOO/UO", "Codice Univoco"
+        };
+
+        File csvFile = filePath.toFile();
+
+        // Using the builder pattern to create the CSV format with headers
+        CSVFormat csvFormat = CSVFormat.Builder.create(CSVFormat.DEFAULT)
+                .setHeader(headers)
+                .setDelimiter(';')
+                .build();
+
+        try (FileWriter writer = new FileWriter(csvFile);
+             CSVPrinter csvPrinter = new CSVPrinter(writer, csvFormat)) {
+
+            // Iterate over each AggregateInstitution object and write a row for each one
+            for (AggregateInstitution institution : institutions) {
+
+                // Determine the value of Codice IPA
+                String codiceIpa = institution.getSubunitType() == null ? institution.getOriginId() : "";
```

Comment:
you could use a more elegant object optional, for ex: `Optional.ofNullable( institution.getSubunitType()).map(ignored -> institution.getOriginId()).orElse("")`